### PR TITLE
Fix RewriteSpectraMap semantics for LoadInstrument

### LINKED ISSUE #14124: LoadInstruments Defaults Cause Major Issues
Now that we have the instrument service, user-editing of IDFs has actually become more complicated, because the instrument loading no-longer routes directly through the `instrument` directory as it always used to. As a result are now telling users to run `LoadInstrument` when they are developing and upgrading their IDFs so they know that the instrument they get is always the instrument def they're working on.

`LoadInstrument` has a Property called `RewriteSpectraMap`. Unfortunately, this defaults to True, and this is causing a lot of confusion because the spectra-detector map gets completely rewritten with it's own custom rules, which bear no link to the originally loaded data. We have recently seen this issue on OFFSPEC.

We want to change the default, but if we do that then we'll end up with nasty down-stream issues, that will be very frustrating to backtrace. We need to force failure and update at the point of use. So the fix is as follows.

* On LoadInstrument Delete the `RewriteSpecraMap` property.
* On LoadInstrument Introduce a new property called `OverwriteSpectraMap`
* Ensure that `OverwriteSpectraMap` is mandatory. 
* Everywhere possible make clear that `OverWriteSpectraMap` is a replacement for `RewriteSpectraMap` in the documentation
* Update everything in the Mantid repo code-base that breaks
* Update the release notes for this major change.

@martyngigg FYI

### PR DESCRIPTION
**Pull request by OwenArnold**
Fixes #14124

**Tester**

Firstly. My apologies, in retrospect I might have separated the introduction of `OptionalBool` from it's usage its introduction and replacement in LoadInstrument. However. This is actually an easy PR to verify.    A code review should be sufficient, because if the replacement was not working, it would be very apparent from the automated tests.

If you feel it necessary to run something manually. You could try to run LoadInstrument without providing the RewriteSpectraMap property value, you'd see it choke prior to execution.

### CODE REVIEW

File: Framework/Kernel/test/PropertyManagerTest.h [hunk 1]
```diff
@@ -481,6 +481,15 @@ class PropertyManagerTest : public CxxTest::TestSuite {
     TS_ASSERT_EQUALS(i, 655);
   }
 
+  void test_char_array() {
```
Comment (martyngigg) [inline Framework/Kernel/test/PropertyManagerTest.h]: Is there a missing `TS_ASSERT` here - It looks like the test doesn't actually do anything? Is this supposed to test whether leaving it blank raises an exception?

Comment (OwenArnold) [inline Framework/Kernel/test/PropertyManagerTest.h]: I agree. I have no memory of doing this, but it should be removed.

Comment (martyngigg) [inline Framework/Kernel/test/PropertyManagerTest.h]: Actually I think it's a good test if the proper asserts are added. I hadn't seen a test that tested it throws in this situation?

Comment (OwenArnold) [inline Framework/Kernel/test/PropertyManagerTest.h]: @martyngigg Test put in place

**Comment by martyngigg** [comment]
Is there a test for a declared property with an optional bool that throws when unset?

**Comment by martyngigg** [comment]
Thanks for adding that. It looks good now.

**Comment by OwenArnold** [comment]
@martyngigg Thanks

**Pull request merged**